CannibalisticZombies/Assets/Scripts/HealthUI.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class HealthUI : MonoBehaviour
+{
+    // HealthUI displays current health.
+    // Calling SetHealth() with new Damage amount update healthBar to that amount.
+    public Image healthBar;
+    public float currentHealth;//unnecessary in implementation
+    public float animationSpeed; //Rate at which damage bar fills/empties per frame
+    public Color damageColor=Color.red;//Color of bar when take damage
+    public Color healColor=Color.green; //Color of bar when heals/removes damage
+    private float fillTarget;//Decimal amount of damage bar to be filled
+
+
+
+    // Update is called once per frame
+    void Update()
+    {
+        TestMethod(); // Remove in actual implementation
+        UpdateHealth();
+    }
+
+    public void SetHealth(float newHealth, float maxHealth = 100f)
+    {
+        //newHealth is the newHealth to be displayed
+        //maxHealth is maximum health, defaults to 100
+        fillTarget = 1- newHealth / maxHealth;
+        fillTarget = Mathf.Clamp(fillTarget, 0, 1);
+        currentHealth = Mathf.Clamp(newHealth,0,maxHealth);
+        Debug.Log(fillTarget);
+        Debug.Log(healthBar.fillAmount);
+
+    }
+
+    private void TestMethod()
```

Comment:
For **every** class and method, we want to put a comment block at the top. We can use it to define the method, but even if we don't put any information it makes the code much easier to read.

Ex:
```
///-/////////////////////////////////////////////////////////////////////
///
/// This class defines the class blah blah blah
/// 
public class MyClass {

   ///-/////////////////////////////////////////////////////////////////////
   ///
   public void Start {
   
   }
}
```